samples/cpp/qrcode.cpp

```diff
@@ -157,7 +160,7 @@ void drawQRCodeResults(Mat& frame, const vector<Point>& corners, const vector<cv
 
 static
 void runQR(
-    QRCodeDetector& qrcode, const Mat& input,
+    QRCodeDetectorBase& qrcode, const Mat& input,
```

Comment:
Use `const reference` for inputs

Reply:
added `const reference` and added const to `decode` methods